crates/uv-resolver/src/resolver/environment.rs

```diff
@@ -374,15 +378,62 @@ impl ResolverEnvironment {
     /// This is useful in contexts where one wants to display a message
     /// relating to a particular fork, but either no message or an entirely
     /// different message when this isn't a fork.
-    pub(crate) fn end_user_fork_display(&self) -> Option<impl std::fmt::Display + '_> {
-        match self.kind {
+    pub(crate) fn end_user_fork_display(&self) -> Option<String> {
+        match &self.kind {
             Kind::Specific { .. } => None,
-            Kind::Universal { ref markers, .. } => {
-                if markers.is_true() {
-                    None
-                } else {
-                    Some(format!("split ({markers:?})"))
+            Kind::Universal {
+                initial_forks: _,
+                markers,
+                include,
+                exclude,
+            } => {
+                let format_conflict_item = |conflict_item: &ConflictItem| {
+                    format!(
+                        "{}{}",
+                        conflict_item.package(),
+                        match conflict_item.conflict() {
+                            ConflictPackage::Extra(extra) => format!("[{extra}]"),
+                            ConflictPackage::Group(group) => {
+                                format!("[group:{group}]")
+                            }
+                        }
+                    )
+                };
+
+                if markers.is_true() && include.is_empty() && exclude.is_empty() {
```

Comment:
We were previously treating forks with a true marker as non-distinct, even if they were conflict forks.